pytorch_lightning/core/lightning.py

```diff
@@ -1610,14 +1610,24 @@ def tbptt_split_batch(self, batch, split_size):
 
         return splits
 
-    def summarize(self, mode: Optional[str] = ModelSummary.MODE_DEFAULT) -> Optional[ModelSummary]:
+    def summarize(self, mode: Optional[str] = "top", max_depth: Optional[int] = None) -> Optional[ModelSummary]:
         model_summary = None
 
-        if mode in ModelSummary.MODES:
-            model_summary = ModelSummary(self, mode=mode)
-            log.info("\n" + str(model_summary))
-        elif mode is not None:
-            raise MisconfigurationException(f"`mode` can be None, {', '.join(ModelSummary.MODES)}, got {mode}")
+        if max_depth is None:
+            rank_zero_deprecation(
+                "The use of model.summarize(mode) argument will be "
+                "deprecated in a future release, in favour of 'max_depth'."
```

Comment:
This should mention when it was deprecated and when it will be removed

See for example another deprecation PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/7168/files

Reply:
Which versions do you want to target for deprecation / removal?  v1.4 and  v1.5?

"The use of `model.summarize(mode)` argument is deprecated in v1.4 and will be removed in v1.5, use `max_depth` argument instead"